distribution/pull_v2_unix.go

```diff
@@ -5,9 +5,14 @@ package distribution
 import (
 	"github.com/docker/distribution"
 	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/docker/distribution/xfer"
 )
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
+	if ld.src.MediaType == schema2.MediaTypeForeignLayer {
```

Comment:
> it also stops the useless retries, which make it seem like a temporary error rather than a platform compatibility issue

This might the real bug: are these errors not being marked as terminal?